conf/machine/include/imx-base.inc

```diff
@@ -656,11 +656,19 @@ WKS_FILE_DEPENDS:append:imx-generic-bsp:mx9-generic-bsp = " \
     ${@oe.utils.ifelse(d.getVar('UBOOT_PROVIDES_BOOT_CONTAINER') == '0', 'imx-boot', '')} \
 "
 
-SOC_DEFAULT_WKS_FILE ?= "imx-uboot-bootpart.wks.in"
-SOC_DEFAULT_WKS_FILE:mx8-generic-bsp ?= "imx-imx-boot-bootpart.wks.in"
+SOC_DEFAULT_WKS_FILE                 ?= "imx-uboot-bootpart.wks.in"
 SOC_DEFAULT_WKS_FILE:mxs-generic-bsp ?= "imx-uboot-mxs-bootpart.wks.in"
+SOC_DEFAULT_WKS_FILE:mx8-generic-bsp ?= "imx-imx-boot-bootpart.wks.in"
 SOC_DEFAULT_WKS_FILE:mx9-generic-bsp ?= "imx-imx-boot-bootpart.wks.in"
 
+WIC_BOOTLOADER = " \
+    ${@bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'imx-boot',   'imx-boot-tagged', \
+       bb.utils.contains('IMX_DEFAULT_BOOTLOADER', 'u-boot-imx', '${WIC_BOOTLOADER_32BIT}', \
```

Comment:
I realize that `PREFERRED_PROVIDER_u-boot` and `PREFERRED_PROVIDER_virtual/bootloader` and be overridden, so should I use one of these here instead of `IMX_DEFAULT_BOOTLOADER`?

Reply:
I think it must use an `imx-boot` prefix to decide what to use. Users might use different `IMX_DEFAULT_BOOTLOADER`, and we should respect that.

I am wondering how to do this in a generic way...